eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/FluentMethodNameCheck.java

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Model Class Method:
+ *  Fluent Methods: All methods that return an instance of the class, and that have one parameter.
+ *  The method name should not start with {@code avoidStartWords}.
+ */
+public class FluentMethodNameCheck extends AbstractCheck {
+
+    private static final String FLUENT_METHOD_ERR = "\"%s\" fluent method name should not start with keyword \"%s\".";
+
+    private Set<String> avoidStartWords = new HashSet<>();
+    private String className;
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.METHOD_DEF
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.CLASS_DEF:
+                className = ast.findFirstToken(TokenTypes.IDENT).getText();
+                break;
+            case TokenTypes.METHOD_DEF:
+                checkMethodNameStartWith(ast);
+                break;
+        }
+    }
+
+    /**
+     * Log the error if the method name is not start with {@code avoidStartWord}
+     * @param ast METHOD_DEF AST node
+     */
+    private void checkMethodNameStartWith(DetailAST ast) {
+        // 1, parameter count should be 1
+        Optional<DetailAST> parametersASTOption = TokenUtil.findFirstTokenByPredicate(ast, c -> c.getType() == TokenTypes.PARAMETERS && c.getChildCount() == 1);
+        if (parametersASTOption.isPresent()) { // one param method
```

Comment:
/*
 * Finds a fluent method by checking for the following conditions:
 * 1. Method has a single parameter.
 * 2. Method return type is itself.
 */
public void checkMethodName(DetailAST ast) {
    if (!parametersAST.isPresent()) {
        return;
    }

    if (!typeAST.isPresent()) {
        return;
    }
}